OtripleS.Web.Api.Tests.Unit/Services/Foundations/Attendances/AttendanceServiceTests.Validations.Add.cs

```diff
@@ -60,9 +60,10 @@ public async void ShouldThrowValidationExceptionOnCreateWhenAttendanceIsNullAndL
             string invalidText)
         {
             // given
-            var invalidStuent = new Attendance
+            var invalidAttendance = new Attendance
             {
-                Notes = invalidText
+                Notes = invalidText,
+                Status = AttendanceStatus.Absent
```

Comment:
This should be a part of a different kind of validation test.

Reply:
With my knowledge limitation, I failed to determine what to do as I've modified following the existing references in other `Enum Validation Tests`. If possible, I ask for your explicit suggestion!

Reply:
So, basic structural validations have their own test. This is the one we are pointing at here in this comment section. Basic structural validation doesn't try to consider a `valid` structural value to be invalid based on some logic.
For instance, structural validations are concerned with a string being `null`, `empty` or `whitespace`. But if a string say a student name is less than 3 characters. Your structural validations and therefore this basic test won't bother with that because it belongs to it's own test.

Now, apply that to the enum validations. with enums you have to write two different tests. The first, is to ensure the enum is not out of range. for instance, if you have an enum `A`, `B` and `C`. If the incoming value has value `3` which is outside of the default assigned range of values for the aforementioned enum values then we should throw `InvalidStudentException` referring to that very enum's property on the model.

The second test is to validate that an incoming new entity may not have `B` or `C` as a value. That's called Logical Validation. it belong to it's own unit test method and shall not be a part of the basic structural validation tests here.

I hope this makes sense.

Finally, I strongly urge you to read the `2.1.3.1 Validations` section in The Standard [here](https://github.com/hassanhabib/The-Standard/blob/master/2.%20Services/2.1%20Foundations/2.1%20Foundations.md#2131-validation) to better understand the difference between these validation types.

Reply:
Thanks, @hassanhabib  for your time & the suggestion.
I'll go through the instructed reading soon and get here with related queries if any!

Reply:
@arafat-pro awesome. thanks.